engine/table/build.gradle

```diff
@@ -106,16 +106,22 @@ test {
 
 // The Parallel tests are now functionally equivalent to the Serial test logic
 // TODO (deephaven-core#643): Fix "leaking" parallel tests
-TestTools.addEngineTest(project, 'Parallel', false, false, true)
+
+def allowParallel = Boolean.parseBoolean(findProperty('allowParallel') as String ?: 'false')
+
+TestTools.addEngineTest(project, 'Parallel', allowParallel, false, !allowParallel)
 // testSerial: non-parallel, not a engine test, isolated
-TestTools.addEngineTest(project, 'Serial', false, false, true)
+TestTools.addEngineTest(project, 'Serial', allowParallel, false, true)
 // testOutOfBand: non-parallel, not a engine test, not isolated
-TestTools.addEngineTest(project, 'OutOfBand', false, false, false)
+TestTools.addEngineTest(project, 'OutOfBand', allowParallel, false, false)
```

Comment:
There are a few other projects that call TestTools.addEngineTest - do we want to give them the same treatment? If so, I wonder if it makes sense to move some of this logic into simpler entry points: `TestTools.addEngineParallelTest(project)`, `TestTools.addEngineSerialTest(project)`, etc.

Also - I know there were many salient details, some of which may be documented, (but that I can't exactly remember right now) as to why the properties (at least in DHE) had to be exactly the way they were.

As part of our #1759 issue, there may need to be significant re-working of how we handle this test structuring.